lib/fluent/plugin/in_tail.rb

```diff
@@ -117,6 +114,30 @@ def configure_tag
       end
     end
 
+    def configure_encoding
+      if @encoding
+        @encoding = parse_encoding_param(@encoding)
+        if @from_encoding
+          @from_encoding = parse_encoding_param(@from_encoding)
+        end
+      else
+        if @from_encoding
+          $log.warn "'from_encoding' parameter must be specified in conjunction with 'encoding' paramter."
+          $log.warn "these parameter set to nil."
+          @encoding = nil
+          @from_encoding = nil
+        end
+      end
+    end
+
+    def parse_encoding_param(encoding_name)
+      begin
+        Encoding.find(encoding_name) if encoding_name
```

Comment:
It seems that it's invalid to call this method with `nil` argument, and there are no code to do such things.
So `if` in postposition looks nonsense.